roles/openstack-setup/tasks/images.yml

```diff
@@ -1,12 +1,29 @@
 ---
 # NOTE: Glance does not know how to proxy copy-from URLs, so we should instead
 # download the image files locally and create images using the file method.
-- name: download images locally
-  get_url: url={{ item.url }} dest=/tmp/{{ item.filename }}
+- name: create image directory
+  file: path=/opt/stack/glance-images state=directory mode=0755
+
+- name: discover existing images
+  stat:
+    path: /opt/stack/glance-images/{{ item.filename }}
+  run_once: True
+  register: images_downloaded
   with_items: glance.images
+
+- name: download any necessary images
+  get_url:
+    url: "{{ item.0.url }}"
+    dest: "/opt/stack/glance-images/{{ item.0.filename }}"
+  run_once: True
+  when: item.1
```

Comment:
I'm not sure if that will work, or if you need to tack on `item.1.exists`. Will watch the test output.